services/director-v2/src/simcore_service_director_v2/core/dynamic_sidecar_settings.py

```diff
@@ -0,0 +1,337 @@
+import logging
+import random
+from enum import Enum, auto
+from pathlib import Path
+
+from models_library.basic_types import BootModeEnum, PortInt
+from models_library.docker import DockerGenericTag
+from models_library.projects_networks import DockerNetworkName
+from models_library.utils.enums import StrAutoEnum
+from pydantic import Field, PositiveFloat, PositiveInt, validator
+from settings_library.base import BaseCustomSettings
+from settings_library.r_clone import RCloneSettings as SettingsLibraryRCloneSettings
+from settings_library.utils_service import DEFAULT_FASTAPI_PORT
+
+from ..constants import DYNAMIC_SIDECAR_DOCKER_IMAGE_RE
+
+_logger = logging.getLogger(__name__)
+
+MINS = 60
+
+
+SERVICE_RUNTIME_SETTINGS: str = "simcore.service.settings"
+SERVICE_REVERSE_PROXY_SETTINGS: str = "simcore.service.reverse-proxy-settings"
+SERVICE_RUNTIME_BOOTSETTINGS: str = "simcore.service.bootsettings"
+
+SUPPORTED_TRAEFIK_LOG_LEVELS: set[str] = {"info", "debug", "warn", "error"}
+
+
+class VFSCacheMode(str, Enum):
+    __slots__ = ()
+
+    OFF = "off"
+    MINIMAL = "minimal"
+    WRITES = "writes"
+    FULL = "full"
+
+
+class EnvoyLogLevel(StrAutoEnum):
+    TRACE = auto()
+    DEBUG = auto()
+    INFO = auto()
+    WARNING = auto()
+    ERROR = auto()
+    CRITICAL = auto()
+
+    def to_log_level(self) -> str:
+        lower_log_level: str = self.value.lower()
+        return lower_log_level
+
+
+class RCloneSettings(SettingsLibraryRCloneSettings):
+    R_CLONE_DIR_CACHE_TIME_SECONDS: PositiveInt = Field(
+        10,
+        description="time to cache directory entries for",
+    )
+    R_CLONE_POLL_INTERVAL_SECONDS: PositiveInt = Field(
+        9,
+        description="time to wait between polling for changes",
+    )
+    R_CLONE_VFS_CACHE_MODE: VFSCacheMode = Field(
+        VFSCacheMode.MINIMAL,
+        description="used primarly for easy testing without requiring requiring code changes",
```

Comment:
there is something fishy in this comment. please adjust. also I don't understand what means "easy testing"???